napari/_vispy/overlays/scale_bar.py

```diff
@@ -82,8 +107,14 @@ def _on_zoom_change(self, *, force: bool = False):
         """Update axes length based on zoom scale."""
 
         # If scale has not changed, do not redraw
-        scale = 1 / self.viewer.camera.zoom
-        if abs(np.log10(self._scale) - np.log10(scale)) < 1e-4 and not force:
+        zoom = self.viewer.camera.zoom
+        scale = 1 / zoom
+        if (
+            # workaround to prevent scale break with last zoom change possible
+            zoom in [501.0, 444.0]
+            or abs(np.log10(self._scale) - np.log10(scale)) < 1e-4
+            and not force
+        ):
```

Comment:
I get a value of `576.0` 🙄 .. I don't know where this comes from.

Reply:
I will remove the validation and will open a new issue with the findings about the odd behavior with the zoom value :+1:

Reply:
Linking the created issue here just in case: https://github.com/napari/napari/issues/6055